src/app/staticanalysis/modules/staticAnalysisCommon.js

```diff
@@ -32,7 +32,7 @@ var basicRegex = {
   CONTRACTTYPE: '^contract ',
   FUNCTIONTYPE: '^function \\(',
   EXTERNALFUNCTIONTYPE: '^function \\(.*\\).* external',
-  CONSTANTFUNCTIONTYPE: '^function \\(.*\\).* constant',
+  CONSTANTFUNCTIONTYPE: '^function \\(.*\\).* \\(constant|view|pure\\)',
```

Comment:
What is this regex doing exactly? Trying to match it based on specifiers? Isn't the AST enough?

Reply:
it check the `Identifier` node (parent of `Function Call` . type attribute contains `function () view`

Reply:
But should it use purely the AST for that?

Reply:
For a function call, it takes a look at the type of the function that is called.

Reply:
This regex is not applied to the source but to the type string.